src/modules/projectConfig/projectConfigFileReader.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License, Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { PathLike } from 'node:fs';
+import { CliFileSystem } from '../../utils/fs-bridge';
+import { ComponentConfig } from '../component';
+import { PackageConfig } from '../package';
+import { ProjectConfig } from './projectConfig';
+import {
+    DEFAULT_CONFIG_FILE_PATH,
+    DEFAULT_CONFIG_LOCKFILE_PATH,
+    DesiredConfigFileComponents,
+    DesiredConfigFilePackages,
+    DesiredConfigFileVariables,
+    VARIABLE_SCOPE_SEPARATOR,
+} from './projectConfigConstants';
+import { ProjectConfigLock } from './projectConfigLock';
+
+interface IProjectConfigReader {
+    read(cliVersion: string, path: PathLike, ignoreLock: boolean): ProjectConfig;
+}
+
+export class MultiFormatConfigReader implements IProjectConfigReader {
+    /**
+     * Checks if a configuration file exists at the given path.
+     * @param path The path to the configuration file.
+     * @returns True if the configuration file exists, otherwise false.
+     */
+    static isAvailable = (path: PathLike = DEFAULT_CONFIG_FILE_PATH) => CliFileSystem.existsSync(path);
+
+    /**
+     * Reads the project configuration using multiple format readers.
+     * @param cliVersion The version of the CLI.
+     * @param path The path to the configuration file.
+     * @param ignoreLock Whether to ignore the project configuration lock file.
+     * @returns The project configuration.
+     * @throws Error if unable to read the configuration file in any format.
+     */
+    static read(cliVersion: string, path: PathLike = DEFAULT_CONFIG_FILE_PATH, ignoreLock: boolean = false): ProjectConfig {
```

Comment:
There should be no static read method in here. Everything should be done in the non static (implemented) read.